File: basic-sample-java/src/main/java/cz/trask/zenid/sample/AppSettings.java
package cz.trask.zenid.sample;

import android.content.Context;
import android.content.SharedPreferences;

public class AppSettings {
    private final SharedPreferences prefs;

    public AppSettings(Context context) {
        prefs = context.getSharedPreferences("zenid_prefs", Context.MODE_PRIVATE);
    }

    public String getServerUrl() { return prefs.getString("server_url", null); }
    public void setServerUrl(String value) {
        if (value == null) prefs.edit().remove("server_url").apply();
        else prefs.edit().putString("server_url", value).apply();
    }

    public String getApiKey() { return prefs.getString("api_key", null); }
    public void setApiKey(String value) {
        if (value == null) prefs.edit().remove("api_key").apply();
        else prefs.edit().putString("api_key", value).apply();
    }

    public String getProfile() { return prefs.getString("profile", ""); }
    public void setProfile(String value) { prefs.edit().putString("profile", value != null ? value : "").apply(); }

    public String getCountry() { return prefs.getString("country", null); }
    public void setCountry(String value) {
        if (value == null) prefs.edit().remove("country").apply();
        else prefs.edit().putString("country", value).apply();
    }

    public String getRole() { return prefs.getString("role", null); }
    public void setRole(String value) {
        if (value == null) prefs.edit().remove("role").apply();
        else prefs.edit().putString("role", value).apply();
    }

    public String getPage() { return prefs.getString("page", null); }
    public void setPage(String value) {
        if (value == null) prefs.edit().remove("page").apply();
        else prefs.edit().putString("page", value).apply();
    }
}

File: basic-sample/src/main/java/cz/trask/zenid/sample/AppSettings.kt
package cz.trask.zenid.sample

import android.content.Context
import android.content.Context.MODE_PRIVATE
import kotlin.reflect.KProperty
import androidx.core.content.edit

class AppSettings(context: Context) {
    private val prefs = context.getSharedPreferences("zenid_prefs", MODE_PRIVATE)

    var serverUrl: String? by NullableStringPref("server_url")
    var apiKey: String? by NullableStringPref("api_key")
    var profile: String by StringPref("profile", "")
    var country: String? by NullableStringPref("country")
    var role: String? by NullableStringPref("role")
    var page: String? by NullableStringPref("page")

    inner class StringPref(private val key: String, private val default: String) {
        operator fun getValue(thisRef: Any?, property: KProperty<*>): String =
            prefs.getString(key, default) ?: default

        operator fun setValue(thisRef: Any?, property: KProperty<*>, value: String) =
            prefs.edit { putString(key, value) }
    }

    inner class NullableStringPref(private val key: String) {
        operator fun getValue(thisRef: Any?, property: KProperty<*>): String? =
            prefs.getString(key, null)

        operator fun setValue(thisRef: Any?, property: KProperty<*>, value: String?) {
            if (value == null) prefs.edit { remove(key) }
            else prefs.edit { putString(key, value) }
        }
    }
}
